tokio/src/future/maybe_done.rs

```diff
@@ -0,0 +1,76 @@
+//! Definition of the MaybeDone combinator
+
+use std::future::Future;
+use std::mem;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+/// A future that may have completed.
+#[derive(Debug)]
+pub(crate) enum MaybeDone<Fut: Future> {
```

Comment:
Instead of dealing w/ refactoring internals to remove dependency on the few futures-util helpers, I copied them here as a temporary measure.